### ISSUE TITLE
TSC crashed - JsErrorScriptException 0x30001

### ISSUE THREAD

**Issue by pankleks** [Microsoft/TypeScript#20765]
When building project I have tsc crashing - all info is:

```
Script failed with error: 'JsErrorScriptException (0x30001)'.
Debug Failure. type must be class or interface
Error: Debug Failure. type must be class or interface
   at fail (Unknown script code:2156:13)
   at getBaseTypes (Unknown script code:22934:21)
   at getSuperClass (Unknown script code:30890:13)
   at isPropertyDeclaredInAncestorClass (Unknown script code:30879:17)
   at checkPropertyNotUsedBeforeDeclaration (Unknown script code:30852:13)
   at checkPropertyAccessExpressionOrQualifiedName (Unknown script code:30827:13)
   at checkPropertyAccessExpression (Unknown script code:30799:13)
   at checkExpressionWorker (Unknown script code:33312:21)
   at checkExpression (Unknown script code:33263:17)
   at checkNonNullExpression (Unknown script code:30784:13)
```

TSC 2.6.2

**Comment by pankleks** [comment]
Actually I found solution - adding `<reference path="..` solved the issue.
However still - I think it could get better error ...

I had `class Foo` defined in file 1.
In file 2 I had:
```ts
export class UseFoo {
  private static _foo = new Foo();
```

This crashed with above error.

If I changed it to:
```ts
const _foo = new Foo();
export class UseFoo {
  ...
```

I got error - class Foo is used before it's declaration - which guided me to use `<reference ...`
When reference was added, original code (with static) worked ok.
